website/guides/advanced/schemas.md.erb

```diff
@@ -0,0 +1,319 @@
+{ "id": "user2", "gdpr": false, "email": "us-user2@timber.io" }
+{ "id": "user3", "gdpr": true, "email": "eu-user3@timber.io" }
+```
+
+In our theoretical product, we're expanding into the EU and want to comply with the GDPR. In our case, that means our
+application can't send EU user data to our US based kafka. (We're not lawyers, this is not a magic GDPR-compliance
+config, just a little example!)
+
+We can build a config that will do the first part of this, but we'll just output to console for ease of this example.
+
+```toml title="vector.toml"
+data_dir = "./data"
+dns_servers = []
+
+[sources.application]
+max_length = 102400
+type = "stdin"
+
+[transforms.parse]
+inputs = ["application"]
+drop_field = true
+drop_invalid = false
+type = "json_parser"
+
+[transforms.not_gdpr]
+type = "filter"
+inputs = ["parse"]
+condition.type = "check_fields"
+condition."gdpr.eq" = "false"
+
+[transforms.gdpr_to_strip]
+type = "filter"
+inputs = ["parse"]
+condition.type = "check_fields"
+condition."gdpr.eq" = "true"
+
+[transforms.gdpr_stripped]
+type = "remove_fields"
+inputs = ["gdpr_to_strip"]
+fields = ["email"]
+
+[sinks.console]
+healthcheck = true
+inputs = ["not_gdpr", "gdpr_stripped"]
+type = "console"
+encoding = "json"
+[sinks.console.buffer]
+type = "memory"
+max_events = 500
+when_full = "block"
+```
+
+Let's have a look:
+
+```bash
+$ cat <<-EOF | cargo run -- --config test.toml
+{ "id": "user1", "gdpr": false, "email": "us-user1@timber.io" }
+{ "id": "user2", "gdpr": false, "email": "us-user2@timber.io" }
+{ "id": "user3", "gdpr": true, "email": "eu-user3@timber.io" }
+EOF
+Feb 05 16:13:59.241  INFO source{name=application type=stdin}: vector::sources::stdin: finished sending
+{"id":"user1","timestamp":"2020-02-06T00:13:59.241801798Z","host":"obsidian","email":"us-user1@timber.io","gdpr":false}
+{"gdpr":false,"host":"obsidian","email":"us-user2@timber.io","timestamp":"2020-02-06T00:13:59.241815255Z","id":"user2"}
+{"id":"user3","gdpr":true,"host":"obsidian","timestamp":"2020-02-06T00:13:59.241816010Z"}
+Feb 05 16:15:27.945  INFO vector: Shutting down.
+```
+
+Don't know where events are coming from? You can use the `geoip` transform an `ipv4` field and get a grip on that!
+
+
+## Sink field filtering
+
+While it's often reasonable to remove this kind of data at the pipeline level, we identified use cases that involve
+using values in sinks from these fields in sink configuration.
+
+The applications for this include some of the reasons discussed in
+["Pipeline field filtering"](#pipeline-field-filtering), but also:
+
+* Stripping off routing related fields
+* Ensuring a specific sink will only ever output specific fields (or never output certain fields)
+
+
+### Example: Per host kafka topics
+
+Lets take a look at what that might look like:
+
+```toml title="vector.toml"
+[sinks.output]
+  inputs = ["demo"]
+  type = "kafka"
+
+  # Put events in the host specific topic.
+  topic = "{{service}}"
```

Comment:
Is the topic field templateable? If so, we need to update our docs to reflect that. If not, we should open an issue to make it templateable.

Reply:
Oh, gosh, you're right. It's not templatable.

Reply:
Note this is still an issue!